zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileInfo.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.notebook;
+
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents note file info for storage 
+ *
+ */
+public class FileInfo {
+
+  private ConcurrentMap<String, String> fileName;
+  private boolean forceRename;
+  public static final FileInfo EMPTY = new FileInfo();
+  
+  private FileInfo() {
+    fileName = Maps.newConcurrentMap();
+    forceRename = false;
+  }
+  
+  public static FileInfo createInstance() {
+    return new FileInfo();
+  }
+  
+  public String getFileName(String storageName) {
+    if (!fileName.containsKey(storageName)) {
```

Comment:
no, actually `NotebookReopSync` is unaware of it and each storage layer needs to handle saving into file/renaming. 

So here's how it's handled in this PR, let's say `noteA` is read from local `GitNotebookRepo` with name `noteA.zpln` and it's also saved in `S3NotebookRepo` under same name via sync. when renaming it, those two storages will refer to `note` object to locate the filename and rename it. in case we have one filename, first storage (git) will rename file locally and update filename in `note` object, then when second storage (s3) updating it will refer to `note` object for old filename (in order to rename into new one), however it will contain new filename overwritten by first storage. that's why i added filename per storage so that they don't overwrite each other. In this PR i tried not to change method signatures, and want to handle it storage by storage. once we have this functionality in all storages, we may change some method signatures or put more load into `NotebookRepoSync`, for now i didn't add much functionality in that base class. let me know if it brings some clarity